raytrace.py

```diff
@@ -581,9 +726,16 @@ def next_direction(self, face):
             next_material = None
         return direction, next_material, phenomenon
 
-    def update_energy(self):
+    def update_energy(self, point_1, point_2):
         # TODO: @Ramon
-        pass
+        solid_1 = self.scene.solid_at_point(point_1)
+        solid_2 = self.scene.solid_at_point(point_2)
+        if solid_1 and solid_1 == solid_2:
+            if 'extinction_coefficient' in self.scene.materials[solid_1].properties:
+                k = self.scene.materials[solid_1].properties['extinction_coefficient'](self.wavelength)
+                if k > 0:
+                    d = point_1.distanceToPoint(point_2)
+                    self.energy = self.energy * math.exp(- k * d / 1000.0)
 
```

Comment:
there is no need to take point_1 and point_2 as parameters. You can recover them from self.